api/provider/aws/load_balancer_delete.go

```diff
@@ -62,8 +63,8 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 	}
 
 	securityGroupName := getLoadBalancerSGName(fqLoadBalancerID)
-	securityGroup, err := readSG(l.AWS.EC2, securityGroupName)
-	if err != nil && !strings.Contains(err.Error(), "does not exist") {
+	securityGroup, readSGErr := readSG(l.AWS.EC2, securityGroupName)
```

Comment:
not sure why you are creating a new error here? Seems the pattern has been established to assign it to `err` here and set `err = nil` below if we catch the correct error code. 

It seems really easy to make a mistake by adding `readSGErr` instead of just always assigning to `err`

Reply:
Okay